crates/nu-parser/src/parser.rs

```diff
@@ -4007,11 +4004,14 @@ fn parse_table_expression(working_set: &mut StateWorkingSet, span: Span) -> Expr
         working_set.parse_errors.extend(errs);
         ty
     } else {
-        Type::Table(vec![])
+        Type::table()
     };
 
+    let mut exprs = vec![head.into()];
+    exprs.extend(rows.into_iter().map(Into::into));
+
     Expression {
-        expr: Expr::Table(head, rows),
+        expr: Expr::Table(Table::from_raw(exprs.into())),
```

Comment:
Is not too bad looking at possible reallocs but the API seems less clear now for `ast::Table`
(maybe I am a bit picky for a single primary usage, but the parser... )

Reply:
Yeah, there's only one additional allocation, but this can be eliminated in the future.

Reply:
Alternative to changing it up like in your above comment would be to simply move this into the constructor so the `ast::Table` internals remain hidden.